cli/lib/kontena/cli/stacks/install_command.rb

```diff
@@ -55,7 +56,9 @@ def install_dependencies
 
         cmd << dependency['stack']
         Kontena.run!(cmd)
+        set_env_variables(stack_name, current_grid)
```

Comment:
Not sure if this is actually enough. Should we protect each dep install so that we reset whole ENV back to original? Now each dep can introduce new environment variables that can mess up next dependency (if it resolves variables from env).

Reply:
Right, there are also other envs than just `STACK` and `GRID` that can leak between stacks, if you're explicitly using `to: env: ...` in stack variables.

Not entirely sure if this is necessarily part of this issue though.... this sounds more like #2798, which was fixed before the 1.4 release by not exporting the stack variables to `ENV` by default ( #2802), as well as interpolating values directly from the opto variables instead of from `ENV` (related to #2822).

Reply:
That could be intentional and act as a way to pass stuff from children to parent in some cases. Setting values to local env rarely makes sense otherwise.